tests/unit/jira/test_issues.py

```diff
@@ -1522,3 +1522,39 @@ def test_create_issue_with_labels_as_string(self, issues_mixin: IssuesMixin):
         # Verify result
         assert result.key == "TEST-123"
         assert result.labels == ["bug", "frontend"]
+
+    def test_get_issue_with_config_projects_filter(self, issues_mixin: IssuesMixin):
+        """Test get with projects filter from config."""
+        # Setup mock response
+        mock_issues = {
+            "issues": [
+                {
+                    "id": "10001",
+                    "key": "TEST-123",
+                    "fields": {
+                        "summary": "Test issue",
+                        "issuetype": {"name": "Bug"},
+                        "status": {"name": "Open"},
+                    },
+                }
+            ],
+            "total": 1,
+            "startAt": 0,
+            "maxResults": 50,
+        }
+        issues_mixin.jira.jql.return_value = mock_issues
+        issues_mixin.config.url = "https://example.atlassian.net"
+        issues_mixin.config.projects_filter = "DEV"
+
+        # Mock the API to raise an exception
+        issues_mixin.jira.get_issue.side_effect = Exception("API error")
```

Comment:
This `side_effect` is confusing because if the filter logic works correctly, the `jira.get_issue` method should never be called. The test should focus on verifying that the `ValueError` is raised by the filter logic *before* any API call is attempted.

I recommend removing this line and adding `issues_mixin.jira.get_issue.assert_not_called()` after the `pytest.raises` block to make the test's intent clearer and more precise.

Reply:
Fixed